.github/workflows/publish-autoinstrumentation-java.yaml

```diff
@@ -0,0 +1,44 @@
+name: "Publish Java Auto-Instrumentation"
+
+on:
+  push:
+    paths:
+      - 'autoinstrumentation/java/**'
+      - '.github/workflows/publish-autoinstrumentation-java.yaml'
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2.3.5
+
+      - name: Read version
+        run: echo "VERSION=$(cat autoinstrumentation/java/version.txt)" >> $GITHUB_ENV
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3.6.0
+        with:
+          images: ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-java
```

Comment:
We should publish to quay, like the other images:

https://github.com/open-telemetry/opentelemetry-operator/blob/c3f411cb6f839443354a2039a0d1190d269147bd/.github/workflows/publish-images.yaml#L31-L39

Reply:
Switched to quay. I'd recommend separately considering migrating to GHCR for everything. For an OSS contributor project like OTel I think there is a lot to like about having creds / permissions tied to the repo itself rather than a backchannel

Reply:
Agree. Created https://github.com/open-telemetry/opentelemetry-operator/issues/492